### PR TITLE
Replaced tinydir with rcutils

### PR DESCRIPTION
**Pull request by ahcorde**
This PR https://github.com/ros2/rcutils/pull/218 include some utils in rcutils to replace tinydir

wait for this PR to merge https://github.com/ros2/rcutils/pull/218

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

File: rcl/src/rcl/security_directory.c [hunk 1]
```diff
@@ -23,7 +24,6 @@
 # pragma clang diagnostic push
 # pragma clang diagnostic ignored "-Wembedded-directive"
 #endif
-#include "tinydir/tinydir.h"
```
Comment (mjcarroll) [inline rcl/src/rcl/security_directory.c]: I believe that the surrounding `__clang__` pragmas can be removed now that the `tinydir` include is gone.

**Comment by wjwwood** [comment]
I'll defer to @mjcarroll if that's ok.

File: rcl/src/rcl/security_directory.c [hunk 2]
```diff
@@ -112,12 +104,16 @@ static bool get_best_matching_directory(
...
+    rcutils_file_fini(&file, dir.allocator);
+    if (RCUTILS_RET_OK != rcutils_next_dir(&dir)) {
       goto cleanup;
     }
   }
+
 cleanup:
-  tinydir_close(&dir);
+  if (rcutils_close_dir(&dir) != RCUTILS_RET_OK) {
+    RCL_SET_ERROR_MSG("Error closing the directory");
```
Comment (dirk-thomas) [inline rcl/src/rcl/security_directory.c]: It would be helpful to include the directory path in the error message.

Comment (ahcorde) [reply rcl/src/rcl/security_directory.c]: Included directory path in the error message: https://github.com/ros2/rcl/pull/605/commits/de50b686dc6f8d5fa2963ddee842bdcc89827bca

File: rcl/src/rcl/security_directory.c [hunk 3]
```diff
@@ -112,12 +104,22 @@ static bool get_best_matching_directory(
         memcpy(matched_name, file.name, max_match_length);
       }
     }
-    if (-1 == tinydir_next(&dir)) {
+    rcutils_ret_t ret = rcutils_file_fini(&file, dir.allocator);
+    if (ret != RCUTILS_RET_OK) {
+      RCL_SET_ERROR_MSG_WITH_FORMAT_STRING("%s", rcutils_get_error_string().str);
+      rcutils_reset_error();
```
Comment (dirk-thomas) [inline rcl/src/rcl/security_directory.c]: Aren't the `rcl` error msg functions just a rename of the `rcutils` error msg functions? Isn't the reset effectively resetting the error message set in the line before?

Comment (ahcorde) [reply rcl/src/rcl/security_directory.c]: removed rcutils_reset_error() https://github.com/ros2/rcl/pull/605/commits/f6730a02341c9220677c99e005f314b718f9c947

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9838/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5455/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8001/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9694/)

**Comment by mikaelarguedas** [comment]
Note: https://github.com/ros2/rcl/pull/515 already removes the tinydir dependency completely. This PR will likely conflict with that other one

**Pull request closed without merging**